homeassistant/components/hassio/auth.py

```diff
@@ -67,6 +76,21 @@ def _get_provider(self):
         _LOGGER.error("Can't find Home Assistant auth.")
         raise HTTPNotFound()
 
+
+class HassIOAuth(HassIOBaseAuth):
+    """Hass.io view to handle auth requests."""
+
+    name = "api:hassio:auth"
+    url = "/api/hassio_auth"
+
+    @RequestDataValidator(SCHEMA_API_AUTH)
+    async def post(self, request, data):
+        """Handle auth requests."""
+        self._check_access(request)
+
+        await self._check_login(data[ATTR_USERNAME], data[ATTR_PASSWORD])
```

Comment:
This will raise if the login is not valid. We should catch that and raise HTTPUnauthorized.